tests/unit/job/test_extract.py

```diff
@@ -39,7 +39,7 @@ def test_to_api_repr(self):
         from google.cloud.bigquery import job
 
         config = self._make_one()
-        config.compression = job.Compression.SNAPPY
+        config.compression = job.Compression.SNAPPY.value
```

Comment:
I find this a bit concerning still. Do we need to update the setter to make sure we are storing a string?

Reply:
We COULD:

Note, it appears that we have compression.setters in two different files: 
* [extract.py](https://github.com/googleapis/python-bigquery/blob/d08ca708ba91b01fe5e7095e612c326cd3bcfe98/google/cloud/bigquery/job/extract.py#L50-52)
* [external_config.py](https://github.com/googleapis/python-bigquery/blob/d08ca708ba91b01fe5e7095e612c326cd3bcfe98/google/cloud/bigquery/external_config.py#L725-727)

AND
it appears that for the majority of the setters, we do not do any sort of tests to confirm that we receive the correct type of object beyond confirming if the object is `None`.


```
    @compression.setter
    def compression(self, value):
        # presume a stringify-able object will be passed
        self._set_sub_prop("compression", str(value))
    
    # OR
    
        # explicitly confirm that a string has been passed
        if isinstance(value, str):
            self._set_sub_prop("compression", value)
        else:
            raise TypeError(f"Compression should be passed a `str` object, received a `{type(value}` instead.")
        
```

Reply:
Good point. Ideally we keep that simple and consistent. The main thing I want us to confirm is the JSON we send to the REST API is unchanged.

Maybe instead of updating the setter, we call [json.dumps](https://docs.python.org/3/library/json.html#json.dumps) with `sort_keys=True` for determinism in the resource comparison, below.

Reply:
With `json.dumps` below, we shouldn't need to do `.value`

```suggestion
        config.compression = job.Compression.SNAPPY
```